File: favourite-lyrics-rest/src/main/java/org/jesperancinha/lyrics/rest/LyricsController.java
package org.jesperancinha.lyrics.rest;

import org.jesperancinha.lyrics.domain.data.LyricsDto;
import org.jesperancinha.lyrics.domain.data.LyricsFullDto;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;

import java.util.List;
import java.util.UUID;

public interface LyricsController {

    @PostMapping("/lyrics")
    ResponseEntity<Void> addLyrics(
            @RequestBody
                    LyricsDto lyricsDto);

    @DeleteMapping("/lyrics")
    ResponseEntity<String> removeLyrics(
            @RequestBody
                    LyricsDto lyricsDto);

    @PutMapping("/lyrics")
    ResponseEntity<String> updateLyrics(
            @RequestBody
                    LyricsDto lyricsDto);

    @GetMapping("/lyrics/{lyricsId}")
    ResponseEntity<LyricsDto> getLyricsById(
            @PathVariable
                    UUID lyricsId);

    @GetMapping("/lyrics")
    ResponseEntity<List<LyricsDto>> lyrics();

    @GetMapping("/complete/lyrics")
    ResponseEntity<List<LyricsFullDto>> complete();

    @GetMapping("/lyrics/random")
    ResponseEntity<LyricsDto> getRandomLyric();

}

File: favourite-lyrics-rest/src/main/kotlin/org/jesperancinha/lyrics/rest/LyricsController.kt
package org.jesperancinha.lyrics.rest

import org.jesperancinha.lyrics.domain.data.LyricsDto
import org.jesperancinha.lyrics.domain.data.LyricsFullDto
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import java.util.*

interface LyricsController {
    @PostMapping("/lyrics")
    fun addLyrics(
        @RequestBody lyricsDto: LyricsDto?
    ): ResponseEntity<Void>

    @DeleteMapping("/lyrics")
    fun removeLyrics(
        @RequestBody lyricsDto: LyricsDto?
    ): ResponseEntity<String>

    @PutMapping("/lyrics")
    fun updateLyrics(
        @RequestBody lyricsDto: LyricsDto?
    ): ResponseEntity<String>

    @GetMapping("/lyrics/{lyricsId}")
    fun getLyricsById(
        @PathVariable lyricsId: UUID?
    ): ResponseEntity<LyricsDto?>

    @GetMapping("/lyrics")
    fun lyrics(): ResponseEntity<List<LyricsDto>>

    @GetMapping("/lyrics/complete")
    fun complete(): ResponseEntity<List<LyricsFullDto>>

    @get:GetMapping("/lyrics/random")
    val randomLyric: ResponseEntity<LyricsDto?>
}